Path pattern object value matcher does not handle backticks

I know there's a default CVA callee matcher that's supposed to handle the `compoundVariants` property 

https://github.com/schoero/eslint-plugin-better-tailwindcss/blob/7a61d18f13ae67c504652d59d31d7deda57d881a/src/options/callees/cva.ts#L28-L31

However this doesn't seem to work when the code is using backticks

For example
```ts
const style = cva("px-4", {
  variants: {
    myVariant: {
      a: "",
      b: "",
    },
  },
  compoundVariants: [
    {
      class: "m-1",
      myVariant: `a`, // << backtick in the variant value
    },
  ],
});
```
will produce a lint error
```
/project/workspace/src/App.tsx
  14:19  error  Unregistered class detected: a              better-tailwindcss/no-unregistered-classes
```

Whereas a double quote will be fine
```ts
const style = cva("px-4", {
  variants: {
    myVariant: {
      a: "",
      b: "",
    },
  },
  compoundVariants: [
    {
      class: "m-1",
      myVariant: "a", // << double quote variant value
    },
  ],
});
```

I can also reproduce it in test by changing

https://github.com/schoero/eslint-plugin-better-tailwindcss/blob/7a61d18f13ae67c504652d59d31d7deda57d881a/src/utils/matchers.test.ts#L24-L31

to use a template literal
```ts
      const code = `const obj = {
        root: {
          nested: {
            value: \`value\`
          }
        }
      };`;
```

```
 FAIL  src/utils/matchers.test.ts > matchers > getObjectPath > should return the correct object path for values
AssertionError: expected undefined to be 'root.nested.value' // Object.is equality

- Expected: 
"root.nested.value"

+ Received: 
undefined

 ❯ src/utils/matchers.test.ts:69:22
     67| 
     68|         const path = getESObjectPath(value);
     69|         expect(path).toBe("root.nested.value");
       |                      ^
     70| 
     71|       });
```

I had a look and the problem seems to be the `getESObjectPath` function doesn't seem to handle a `node.type` of `TemplateElement` 

https://github.com/schoero/eslint-plugin-better-tailwindcss/blob/7a61d18f13ae67c504652d59d31d7deda57d881a/src/parsers/es.ts#L369-L381

Adding in
```diff
  if(
    node.type !== "Property" &&
    node.type !== "ObjectExpression" &&
    node.type !== "ArrayExpression" &&
    node.type !== "Identifier" &&
--  node.type !== "Literal"
++  node.type !== "Literal" &&
++  node.type !== "TemplateElement"
  ){
    return;
  }
```
fixes it but introduces a type error below

```ts
  if(node.parent.type === "ArrayExpression" && node.type !== "Property"){
    const index = node.parent.elements.indexOf(node);
```

--- a/src/parsers/es.ts
+++ b/src/parsers/es.ts
@@ -375,7 +375,8 @@ export function getESObjectPath(node: ESNode & Partial<Rule.NodeParentExtension>
     node.type !== "ObjectExpression" &&
     node.type !== "ArrayExpression" &&
     node.type !== "Identifier" &&
-    node.type !== "Literal"
+    node.type !== "Literal" &&
+    node.type !== "TemplateElement"
   ){
     return;
   }
@@ -405,7 +406,7 @@ export function getESObjectPath(node: ESNode & Partial<Rule.NodeParentExtension>
     return getESObjectPath(property);
   }
 
-  if(node.parent.type === "ArrayExpression" && node.type !== "Property"){
+  if(node.parent.type === "ArrayExpression" && node.type !== "Property" && node.type !== "TemplateElement"){
     const index = node.parent.elements.indexOf(node);
     paths.unshift(`[${index}]`);
   }

--- a/src/utils/matchers.test.ts
+++ b/src/utils/matchers.test.ts
@@ -70,6 +70,29 @@ describe("matchers", () => {
 
       });
 
+      it("should return the correct object path for template literal values", () => {
+
+        const code = `const obj = {
+          root: {
+            nested: {
+              value: \`value\`
+            }
+          }
+        };`;
+
+        const ast = withParentNodeExtension(parse(code, { ecmaVersion: "latest" }) as ESNode);
+
+        const value = findNode(ast, (node): node is ESNode => {
+          return isESNode(node) && isESStringLike(node) && isInsideObjectValue(node);
+        });
+
+        assert(value);
+
+        const path = getESObjectPath(value);
+        expect(path).toBe("root.nested.value");
+
+      });
+
       it("should put names in quotes if they are not valid identifiers", () => {
 
         const code = `const obj = {

